### PR TITLE
chore: upgrade go version to 1.22.4

### PR DESCRIPTION
**Pull request by zhiying-lin**
### Description of your changes
https://github.com/Azure/fleet/actions/runs/9574800698/job/26398647455
Fix CVE-2024-24790
┌─────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version  │                           Title                            │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-2[47](https://github.com/Azure/fleet/actions/runs/9574800698/job/26398647455#step:8:48)90 │ CRITICAL │ fixed  │ 1.22.3            │ 1.21.11, 1.22.4 │ golang: net/netip: Unexpected behavior from Is methods for │
│         │                │          │        │                   │                 │ IPv4-mapped IPv6 addresses                                 │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-24790                 │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴────────────────────────────────────────────────────────────┘

upgrade go version when building images

Fixes #

I have:

- [ ] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: .github/workflows/trivy.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ env:
   MEMBER_AGENT_IMAGE_NAME: member-agent
   REFRESH_TOKEN_IMAGE_NAME: refresh-token
 
-  GO_VERSION: '1.22.3'
+  GO_VERSION: '1.22.4'
```
Comment (ryanzhang-oss) [inline .github/workflows/trivy.yml:20]: why does this matter?

Comment (zhiying-lin) [reply .github/workflows/trivy.yml:20]: it should not, keep it consistent with the go version when building the images.

Comment (ryanzhang-oss) [reply .github/workflows/trivy.yml:20]: but our go mod is 1.22.3

Comment (zhiying-lin) [reply .github/workflows/trivy.yml:20]: yeah, it should not matter though. I'm fine to keep it consistent with our go mod file. Either should be fine.

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**